notifier/tests/test_tasks.py

```diff
@@ -168,6 +197,135 @@ def test_generate_and_send_digests_retry_cs(self):
+
+    def test_generate_and_send_digests_flagged_retry_ses(self):
+        """
+        Test that retries are attempted
+        """
+        messages = make_messages()
+
+        # setting this here because override_settings doesn't seem to
+        # work on celery task configuration decorators
+        expected_num_tries = 1 + settings.FORUM_DIGEST_TASK_MAX_RETRIES
+        mock_backend = Mock(name='mock_backend', send_messages=Mock(
+            side_effect=SESMaxSendingRateExceededError(400, 'Throttling')))
+        with patch('notifier.connection_wrapper.dj_get_connection', return_value=mock_backend):
+            # execute task - should fail, retry twice and still fail, then
+            # give up
+            try:
+                generate_and_send_digests_flagged.delay(messages)
+            except SESMaxSendingRateExceededError as e:
+                self.assertEqual(
+                    mock_backend.send_messages.call_count,
+                    expected_num_tries,
+                )
+
+    @override_settings(FORUM_DIGEST_TASK_BATCH_SIZE=9)
+    def test_do_forums_digests_flagged(self):
+        """
+        Test that we can send forum digests for flagged posts
+        """
+        test_threads = (
+            {
+                'course_id': 'org/course/run',
+                'commentable_id': TEST_COMMENTABLE,
+                'comment_thread_id': i,
+            }
+            for i in xrange(1, 6)
+        )
+        test_moderators = (usern(i) for i in xrange(10))
+
+        last_message_batch = [
+            {
+                'course_id': 'org/course/run',
+                'recipient': usern(9),
+                'threads': [
+                    '{}/courses/{}/discussion/forum/{}/threads/{}'.format(
+                        settings.LMS_URL_BASE, 'org/course/run', TEST_COMMENTABLE, i
+                    )
+                    for i in xrange(1, 6)
+                ],
+            }
+        ]
+
+        with patch('notifier.tasks.get_flagged_threads', return_value=test_threads), \
+             patch('notifier.tasks.get_moderators', return_value=test_moderators) as m, \
+             patch('notifier.tasks.generate_and_send_digests_flagged') as t:
+            task_result = do_forums_digests_flagged.delay()
+            self.assertTrue(task_result.successful())
+            m.assert_called_with('org/course/run')
+            self.assertEqual(t.delay.call_count, 2)
+            t.delay.assert_called_with(last_message_batch)
+
+    @override_settings(FORUM_DIGEST_TASK_BATCH_SIZE=10)
+    def test_do_forums_digests_flagged_user_api_unavailable(self):
+        # use list since generator is not reusable for subsequent retries
+        test_threads = [
+            {
+                'course_id': 'org/course/run',
+                'commentable_id': 'test_commentable',
+                'comment_thread_id': i,
+            }
+            for i in xrange(1, 6)
+        ]
```

Comment:
Looks like we're duplicating this logic.
We should split it out into a separate method.